app/uk/gov/hmrc/tradergoodsprofilesdatastore/services/StoreRecordsService.scala

```diff
@@ -19,57 +19,92 @@ package uk.gov.hmrc.tradergoodsprofilesdatastore.services
 import com.google.inject.Inject
 import org.apache.pekko.Done
 import play.api.Logging
-import play.api.mvc.Request
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.tradergoodsprofilesdatastore.connectors.RouterConnector
+import uk.gov.hmrc.tradergoodsprofilesdatastore.models.RecordsSummary.Update
 import uk.gov.hmrc.tradergoodsprofilesdatastore.models.response.Pagination.{recursivePageSize, recursiveStartingPage}
-import uk.gov.hmrc.tradergoodsprofilesdatastore.repositories.RecordsRepository
+import uk.gov.hmrc.tradergoodsprofilesdatastore.repositories.{RecordsRepository, RecordsSummaryRepository}
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class StoreRecordsService @Inject() (routerConnector: RouterConnector, recordsRepository: RecordsRepository)(implicit
+class StoreRecordsService @Inject() (
+  routerConnector: RouterConnector,
+  recordsRepository: RecordsRepository,
+  recordsSummaryRepository: RecordsSummaryRepository
+)(implicit
   ec: ExecutionContext
 ) extends Logging {
 
   def storeRecords(
     eori: String,
     lastUpdatedDate: Option[String]
-  )(implicit request: Request[_], hc: HeaderCarrier): Future[Done] =
-    storeRecordsRecursively(eori, recursiveStartingPage, lastUpdatedDate).flatMap { _ =>
-      routerConnector.getRecords(eori, None, Some(recursiveStartingPage), Some(1)).flatMap { recordsResponse =>
-        val totalRouterRecords = recordsResponse.pagination.totalRecords
-        recordsRepository.getCountWithInactive(eori).flatMap { totalDataStoreRecords =>
-          if (totalRouterRecords == totalDataStoreRecords) {
-            Future.successful(Done)
-          } else {
-            Future.failed(
-              new RuntimeException(
-                s"Data Store and B&T Database are out of sync: There are $totalDataStoreRecords data store records and $totalRouterRecords B&T records."
-              )
-            )
-          }
+  )(implicit hc: HeaderCarrier): Future[Boolean] =
+    storeFirstBatchOfRecords(eori, lastUpdatedDate).flatMap { recordsToStore =>
+      if (recordsToStore > recursivePageSize) {
+        recordsSummaryRepository.set(eori, Some(Update(recursivePageSize, recordsToStore - recursivePageSize))).map {
+          _ =>
+            storeRecordsRecursively(eori, recursiveStartingPage + 1, lastUpdatedDate, 0, recordsToStore).onComplete {
```

Comment:
does the recordsToStore should be before 0, as 0 is recordsStored ?

Reply:
fixed in both places please check